src/solc_0.8/polygon/child/avatar/PolygonAvatarSale.sol

```diff
@@ -0,0 +1,90 @@
+//SPDX-License-Identifier: MIT
+// solhint-disable-next-line compiler-version
+pragma solidity 0.8.2;
+
+import {ECDSAUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/cryptography/ECDSAUpgradeable.sol";
+import {IERC20Upgradeable} from "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import {PolygonAvatarSaleStorage} from "./PolygonAvatarSaleStorage.sol";
+import {IAvatarMinter} from "../../../common/interfaces/IAvatarMinter.sol";
+import {AccessControlUpgradeable} from "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+
+// This contract is final, don't inherit form it.
+contract PolygonAvatarSale is PolygonAvatarSaleStorage {
+    function initialize(
+        IAvatarMinter polygonAvatarAddress_,
+        IERC20Upgradeable sandTokenContractAddress_,
+        address trustedForwarder_,
+        address defaultAdmin_,
+        address storageChanger_
+    ) external initializer {
+        __Context_init_unchained();
+        __ERC165_init_unchained();
+        __AccessControl_init_unchained();
+        __UpgradeableBase_init_unchained(defaultAdmin_, storageChanger_);
+        __EIP712_init_unchained(name, version);
+        __ERC2771Handler_initialize(trustedForwarder_);
+        polygonAvatarAddress = polygonAvatarAddress_;
+        sandTokenContractAddress = sandTokenContractAddress_;
+    }
+
+    function verify(
+        uint8 v,
+        bytes32 r,
+        bytes32 s,
```

Comment:
You could pass only one argument instead of v, r, s

Reply:
if I do that then I must separate them in solidity.